Feat(LIF-1567): add credentials and variables from profiles

danielboloc opened the pull request:
# Overview

- Add `--profile` to all commands

 # JIRA

https://lifebit.atlassian.net/browse/LIF-1566
https://lifebit.atlassian.net/browse/LP-39464
https://lifebit.atlassian.net/browse/LIF-1567
https://lifebit.atlassian.net/browse/LP-37105

# Tests

## Create profile without specifically adding `--profile`

Empty `credentials` and `config`
<img width="395" alt="image" src="https://github.com/user-attachments/assets/6becc88c-3061-40b3-9bb7-52cb3803ebb7" />

Add API (as simple text). If the profile already exists, it reads the current value, masked
<img width="920" alt="image" src="https://github.com/user-attachments/assets/882f27d0-9175-4cd1-8345-7d0cbd695aac" />

Add other variables. If the profile exists, it reads the current value, and it can be modified. Simply pressing "Enter" will leave the value as it is
<img width="956" alt="image" src="https://github.com/user-attachments/assets/4ec14373-8ddc-4081-8ac7-6bddf53ee4ec" />

## Create a named profile (using `--profile`)

<img width="845" alt="image" src="https://github.com/user-attachments/assets/8774f930-b4f4-49bc-a9cd-eb5ba51b5e4e" />

Having already at least 1 profile, now a question to make this new profile the default
<img width="745" alt="image" src="https://github.com/user-attachments/assets/79117164-ffe2-47ce-863e-945b791aa521" />
<img width="785" alt="image" src="https://github.com/user-attachments/assets/91016980-227e-4c31-91c3-f0048b0a2b46" />

## Make a profile the default

Check current profiles (where `default`profile is the default)
<img width="392" alt="image" src="https://github.com/user-attachments/assets/c30dfbb1-2372-4340-9299-d550e136732a" />

Making the `new` profile as default
<img width="1079" alt="image" src="https://github.com/user-attachments/assets/75b5856e-3b4b-4a3b-b941-f015a0f978ff" />
<img width="535" alt="image" src="https://github.com/user-attachments/assets/5ad90110-44dd-4cca-adca-8a45e1f2a9a8" />

## List profiles

<img width="968" alt="image" src="https://github.com/user-attachments/assets/e0f23c61-c34e-4714-ae5a-eb63fed9b40a" />

Now `new` is the default, since we changed before
<img width="387" alt="image" src="https://github.com/user-attachments/assets/2085cc0f-3fe8-4b45-961b-01ae51333534" />

## Remove profile

Remove the `new` profile
<img width="1120" alt="image" src="https://github.com/user-attachments/assets/b9dbbca5-416c-4796-a2c3-1f5b6d7aa0b6" />
<img width="487" alt="image" src="https://github.com/user-attachments/assets/4bb7609b-d43f-450d-b1f5-e40ec164fc43" />

The remaining profile or the `default` profile will become the default
<img width="840" alt="image" src="https://github.com/user-attachments/assets/ce0cf522-036c-42ef-b956-b17b41ab2b7e" />
<img width="382" alt="image" src="https://github.com/user-attachments/assets/3cffd773-03ff-43d2-bed9-77d6fb422423" />

## Use default without `--profile` (in this case `default` profile)

<img width="1247" alt="image" src="https://github.com/user-attachments/assets/9cde1ef9-d4f5-499c-9ab7-c5d1c48171f2" />
<img width="1250" alt="image" src="https://github.com/user-attachments/assets/55715ffb-ce6d-4502-9e07-2232e3dd9096" />
<img width="1360" alt="image" src="https://github.com/user-attachments/assets/48a544ae-7665-488d-a9a3-160bb922c0fe" />


## 🟠 Trying to run with non existent profile

<img width="1249" alt="image" src="https://github.com/user-attachments/assets/04dae285-d321-4496-95ad-8c1e2ccaee7f" />
<img width="1256" alt="image" src="https://github.com/user-attachments/assets/f7d7b722-805d-41a5-b482-b1da7f6905a2" />

dapineyro commented on cloudos_cli/__main__.py:
```diff
@@ -276,9 +383,21 @@ def run(apikey,
+        ssl_cert,
+        profile):
     """Submit a job to CloudOS."""
-    cloudos_url = cloudos_url.rstrip('/')
```
@danielboloc I've tested: you cannot remove this (here and in all the places you removed it). If the user have not configured a profile yet, this is still necessary

danielboloc replied:
I have added it again to all cloudos url variables, thanks for the catch

dapineyro reviewed (changes requested):
Thanks @danielboloc ! This is a huge addition to the repo! I have several comments issues that I've found by testing:
1. If, during configuration, I skip one value using "enter". Instead of leaving that param without default it adds it as "blank". If I use the profile it will use that blank value as it is, giving a nasty misleading error from the API server. I'd expect that, if there is already a default from the CLI (like cloudos URL for instance) I can skip that one and Cloudos-cli will pick the old default from the CLI.
2. If I don't configure a value for a required parameter, say the apikey or the URL, and I use that "blank" value that is created in the profile if I just skip it, we don't have that click parameter check that we used to have for required values and so it fails in a nasty way. This is very confusing for the user
3. when selecting these numerical options (platform executor and repo provider), the number I've selected stays there without any spacing. This is a minor thing to me and could be addressed in a following PR.
<img width="270" alt="Screenshot 2025-05-09 at 16 16 37" src="https://github.com/user-attachments/assets/428f2296-f854-469c-8534-a612f19de491" />
4. Also for these numerical options, skipping them with "enter" when updating, actually overwrites whatever we had with the default (aws and github)
5. I've tested creating first a named profile, then the default one (which I didn't make default). If I want to use `cloudos configure --profile default --make-default` or `cloudos configure --make-default`, in both cases, the message tells me that "default" is already created and ask me to edit it, instead of just make it default
<img width="998" alt="Screenshot 2025-05-09 at 16 20 24" src="https://github.com/user-attachments/assets/adeb025c-3d4b-4523-829e-7bc4cb271d1b" />
6. Cromwell commands do not have the `--profile` option

danielboloc commented:
@dapineyro thank you for the review, here are the new developments

1. As you correctly pointed, when there might be a missing parameter, the error happens way downstream. This is now fixed, for example this profile `write` is missing the `cloudos_url` variable
<img width="458" alt="image" src="https://github.com/user-attachments/assets/c66c4732-58b1-4011-b030-96e21da1d85b" />

running `cloudos job run --profile write --nextflow-profile "test"`, is now taking the default:
<img width="1261" alt="image" src="https://github.com/user-attachments/assets/8cb1661d-a80f-417f-b443-ff3ec8b8be12" />

2. Now required parameters, do generate Click-style error messages

In this case, `--workspace-id` is not defined

<img width="528" alt="image" src="https://github.com/user-attachments/assets/00cb6756-3cc1-49ce-b7a8-b1c376982360" />

Now gives error:
<img width="844" alt="image" src="https://github.com/user-attachments/assets/cc7a8e24-6a7b-4ac9-b006-753953404fc1" />

3. Fixed
4. Fixed
5. Fixed
`cloudos configure --profile no_workspace --make-default`
<img width="515" alt="image" src="https://github.com/user-attachments/assets/a4b034b7-1b05-4911-9045-2d8b79b7f4a7" />
6. Added `--profile` for cromwell command too

dapineyro reviewed (approved):
LGTM!
